src/UI/Areas/Identity/Pages/Account/ResetPasswordConfirmation.cshtml.cs

```diff
@@ -1,10 +1,12 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc.RazorPages;
 
 namespace Microsoft.AspNetCore.Identity.UI.Pages.Account.Internal
 {
+    [AllowAnonymous]
```

Comment:
Users typically reach this after they login | reset password | reset password confirmation...so i think this should not be allowanonymous

Reply:
This is when the user has forgotten the password and clicks the `ForgotPassword` link. The flow you're talking about is through Manage action `ChangePassword`. In the `ForgotPassword` case the user is not logged in and gets an email to reset the password

1. User has to access `ResetPassword` to set a new password (from the link in the email) and is not logged in automatically (has to sign in again)
2. Being redirected to `ResetPasswordConfirmation` after the previous step makes it clear that a new password was set and the user may log in. 
3. If `ResetPasswordConfirmation` is not allowed to be accessed anonymously, the user will be redirected to the Login page without any confirmation.

Reply:
As yes, you are right. I confused myself with Chnage Password flow. Looks good to me then!